client/src/interface.rs

```diff
@@ -806,4 +912,71 @@ impl Stronghold {
     async fn network_actor(&self) -> StrongholdResult<Addr<NetworkActor>> {
         self.registry.send(GetNetwork).await?.ok_or(ActorError::TargetNotFound)
     }
+
+    /// Sync the local stronghold with a remote one.
+    pub async fn sync_with(
+        &self,
+        peer: PeerId,
+        merge_policy: SelectOrMerge<SelectOrMerge<SelectOne>>,
```

Comment:
> Since it's a policy, I would expect some kind of configuration type; maybe a solution to this could be some configuration variants implemented as functions, that returns a specific strategy.

I am not sure if I understand what you mean. Could you draft an example for such a function?

Reply:
Sure. I assume the problem to be solved with synchronization is the currently implied type-hierarchy by Stronghold. You chose an approach to reflect this hierarchy over generic types, which are effectively restricted to `SelectOrMerge` or `SelectOne`, but can potentially anything else. While this is being used to unpack the inner types and match on them, I wonder if a more direct approach, maybe a bit more verbose, would make sense. I'm drafting out an idea, which not not be complete but could give a direction:

The assumption is that synchronization is always an operation on a pair A -> B, where A _wants to synchronize_ with B.

```
/// A specific merge policy type
enum MergePolicy {
    /// This would keep the old entries in A, adding only newer / non-existing ones from B
    Retain,
    
    /// This would update entries in A to B
    Update,
}
``` 

The `MergePolicy` could reside inside a `MergeConfiguration`. 
```
struct MergeConfiguration {
     vaults: MergePolicy,
     records: MergePolicy,
     store: Option<MergePolicy>
}
```

Now a `MergeConfiguration` could be constructed via creation functions: 

```
impl MergeConfigurtion {
    
    /// the function name could definitely be better.
    pub fn update_records_keep_vaults_discard_store() -> Self {
        Self { 
             vaults: MergePolicy::Retain,
             records: MergePolicy::Update,
             store: None
        }
    }
}
```
This way we could bundle the intended synchronization policies for each respective parts while being verbose and clear what part is affected. This would go against your more generic approach, but I would assume that switching the config / policy types would make it easier to reason about.